validation/validation_util.jl

```diff
@@ -34,6 +48,40 @@ function interpolated_mse(reference_time, reference_values, simulation_time,
     return mse
 end
 
+# Calculate the mean relative error (MRE) between interpolated simulation values and reference values
+# over a common time range.
+#
+# Arguments:
+# - `reference_time`    : Time points for the reference data.
+# - `reference_values`  : Data points for the reference data.
+# - `simulation_time`   : Time points for the simulation data.
+# - `simulation_values` : Data points for the simulation data.
+function interpolated_mre(reference_time, reference_values, simulation_time,
+                          simulation_values)
+    if last(simulation_time) > last(reference_time)
+        @warn "simulation time range is larger than reference time range. " *
+              "Only checking values within reference time range."
+    end
+
+    # Remove reference time points outside the simulation time
+    start = searchsortedfirst(reference_time, first(simulation_time))
+    end_ = searchsortedlast(reference_time, last(simulation_time))
+    common_time_range = reference_time[start:end_]
+
+    # Interpolate simulation data at the common time points
+    interpolated_values = [linear_interpolation(simulation_time, simulation_values, t)
+                           for t in common_time_range]
+
+    filtered_values = reference_values[start:end_]
```

Comment:
Can you please extract a function here to avoid duplicate code?
Like
```suggestion
function interpolated_errors(reference_time, reference_values, simulation_time, simulation_values)
    if last(simulation_time) > last(reference_time)
        @warn "simulation time range is larger than reference time range. " *
              "Only checking values within reference time range."
    end

    # Remove reference time points outside the simulation time
    start = searchsortedfirst(reference_time, first(simulation_time))
    end_ = searchsortedlast(reference_time, last(simulation_time))
    common_time_range = reference_time[start:end_]

    # Interpolate simulation data at the common time points
    interpolated_values = [linear_interpolation(simulation_time, simulation_values, t)
                           for t in common_time_range]

    filtered_values = reference_values[start:end_]
    
    return abs.(interpolated_values .- filtered_values)
end
```